### PR TITLE
Add three new action hooks for admin shipping zone methods

### PR DESCRIPTION
**Pull request by robertdevore**
### Changes proposed in this Pull Request:

I had a client project recently where I needed to edit the shipping zone methods screen to add in additional fields, but I noticed there's only one current action hook on the page:

`woocommerce_shipping_zone_before_methods_table`

The new action hooks are:

* woocommerce_shipping_zone_inside_top_methods_table
* woocommerce_shipping_zone_inside_bottom_methods_table
* woocommerce_shipping_zone_after_methods_table

This gives more control over the placement of custom data on the screen, instead of only allowing code to be included before the methods table.

### Changelog entry

> Added three action hooks to the admin shipping zone methods.

### CODE REVIEW

File: includes/admin/settings/views/html-admin-page-shipping-zone-methods.php [hunk 1]
```diff
@@ -97,8 +98,12 @@
 				</table>
 			</td>
 		</tr>
+	<?php do_action( 'woocommerce_shipping_zone_inside_bottom_methods_table' ); ?>
 	</tbody>
```
Comment (claudiulodro) [inline includes/admin/settings/views/html-admin-page-shipping-zone-methods.php]: This looks good. Thanks for your contribution.

One small change please, then this is good to merge: The indenting of the `woocommerce_shipping_zone_inside_bottom_methods_table` and `woocommerce_shipping_zone_inside_top_methods_table` actions is inconsistent. Please indent `woocommerce_shipping_zone_inside_bottom_methods_table` one more tab (so it lines up with the `</tr>` on the previous live).

Comment (robertdevore) [reply includes/admin/settings/views/html-admin-page-shipping-zone-methods.php]: done :+1:

**Review by claudiulodro — APPROVED** [review]

**Review by mikejolley — DISMISSED** [review]
Not a fan of this change.

The 'after' hook is fine, but the one inside the table follows no naming standards from WC (no where uses top + bottom), opens this up to invalid table markup, and goes against how that table works since it's rendered with javascript.

Why are those hooks needed?

**Comment by robertdevore** [comment]
I was working on a client project and they needed to display an estimated delivery time along with each zone they were adding.

The issue was that `woocommerce_shipping_zone_before_methods_table` added the text field above the Title and other default fields.

That's where the top/bottom came in, to help with the text input field placement.

If there's already a way to add/reorder the fields for shipping methods, I couldn't find another way of doing it. 

I can remove those two, and/or alter the naming to fit within a WooCommerce naming style if needed, no problem.

**Comment by mikejolley** [comment]
@robertdevore Would prefer those two were removed. The other hook is ok.

**Comment by robertdevore** [comment]
@mikejolley done :+1:

**Pull request merged**